java/server/src/org/openqa/grid/internal/utils/configuration/GridNodeConfiguration.java

```diff
@@ -0,0 +1,203 @@
+  )
+  public Integer registerCycle;
+
+  @Parameter(
+    names = "-unregisterIfStillDownAfter",
+    description = "<Integer> in ms. If the node remains down for more than unregisterIfStillDownAfter millisec, it will disappear from the hub.Default is 1min."
+  )
+  public Integer unregisterIfStillDownAfter;
+
+  public String getHubHost() {
+    if (hubHost == null) {
+      if (hub == null) {
+        throw new RuntimeException("You must specify either a hubHost or hub parameter");
+      }
+      parseHubUrl();
+    }
+    return hubHost;
+  }
+
+  public Integer getHubPort() {
+    if (hubPort == null) {
+      if (hub == null) {
+        throw new RuntimeException("You must specify either a hubPort or hub parameter");
+      }
+      parseHubUrl();
+    }
+    return hubPort;
+  }
+
+  public String getRemoteHost() {
+    if (remoteHost == null) {
+      if (host == null) {
+        host = "localhost";
+      }
+      if (port == null) {
+        port = 5555;
+      }
+      remoteHost = "http://" + host + ":" + port;
+    }
+    return remoteHost;
+  }
+
+  private void parseHubUrl() {
+    try {
+      URL u = new URL(hub);
+      hubHost = u.getHost();
+      hubPort = u.getPort();
+    } catch (MalformedURLException mURLe) {
+      throw new RuntimeException("-hub must be a valid url: " + hub, mURLe);
+    }
+  }
+
+  private class DesiredCapabilityConverter implements IStringConverter<DesiredCapabilities> {
+    @Override
+    public DesiredCapabilities convert(String value) {
+      DesiredCapabilities capabilities = new DesiredCapabilities();
+      for (String cap : value.split(",")) {
+        String[] pieces = cap.split("=");
+        capabilities.setCapability(pieces[0], pieces[1]);
+      }
+      return capabilities;
+    }
+  }
+
+  public void merge(GridNodeConfiguration other) {
+    super.merge(other);
+    if (other.browser != null) browser = other.browser;
+    if (other.downPollingLimit != null) downPollingLimit = other.downPollingLimit;
+    if (other.hubHost != null) hubHost = other.hubHost;
+    if (other.hubPort != null) hubPort = other.hubPort;
+    if (other.id != null) id = other.id;
+    if (other.nodePolling != null) nodePolling = other.nodePolling;
+    if (other.nodeStatusCheckTimeout != null) nodeStatusCheckTimeout = other.nodeStatusCheckTimeout;
+    if (other.proxy != null) proxy = other.proxy;
+    if (other.register != null) register = other.register;
+    if (other.registerCycle != null) registerCycle = other.registerCycle;
+    if (other.remoteHost != null) remoteHost = other.remoteHost;
+    if (other.unregisterIfStillDownAfter != null) unregisterIfStillDownAfter = other.unregisterIfStillDownAfter;
+  }
+
+  @Override
+  public String toString(String format) {
+    StringBuilder sb = new StringBuilder();
+    sb.append(super.toString(format));
+    // id, remoteHost are not command line parameters, excluding from this?
```

Comment:
except the toString method right now is used on some of the grid console pages to show what arguments are used to run this? suppose it won't hurt to have more output